segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/AttributeMixer.java

```diff
@@ -0,0 +1,149 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.server.containers;
+
+import com.google.common.base.Preconditions;
+import io.pravega.segmentstore.contracts.Attributes;
+import io.pravega.segmentstore.server.AttributeIterator;
+import io.pravega.segmentstore.server.SegmentMetadata;
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+import lombok.NonNull;
+import lombok.val;
+
+/**
+ * Combines Segment Attributes from multiple sources.
+ */
+class AttributeMixer implements AttributeIterator {
```

Comment:
I'm not exactly sure what this mixer is supposed to do. Is it creating an iterator that merges attributes in Tier 2 with attributes (more recently) updated in memory? Or is it related to core vs. extended attributes?

Reply:
Renamed this to `SegmenAttributeIterator` and added more Javadoc